features/series/src/main/java/com/chesire/nekome/app/series/collection/ui/CollectionViewModel.kt

```diff
@@ -0,0 +1,211 @@
+@file:OptIn(ExperimentalCoroutinesApi::class)
+
+package com.chesire.nekome.app.series.collection.ui
+
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.chesire.nekome.app.series.R
+import com.chesire.nekome.app.series.collection.core.CollectSeriesUseCase
+import com.chesire.nekome.app.series.collection.core.CurrentFiltersUseCase
+import com.chesire.nekome.app.series.collection.core.CurrentSortUseCase
+import com.chesire.nekome.app.series.collection.core.FilterSeriesUseCase
+import com.chesire.nekome.app.series.collection.core.IncrementSeriesUseCase
+import com.chesire.nekome.app.series.collection.core.RefreshSeriesUseCase
+import com.chesire.nekome.app.series.collection.core.ShouldRateSeriesUseCase
+import com.chesire.nekome.app.series.collection.core.SortSeriesUseCase
+import com.chesire.nekome.app.series.collection.core.UpdateFiltersUseCase
+import com.chesire.nekome.app.series.collection.core.UpdateSortUseCase
+import com.chesire.nekome.core.flags.SeriesType
+import com.chesire.nekome.core.flags.SortOption
+import com.github.michaelbull.result.onFailure
+import com.github.michaelbull.result.onSuccess
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+
+// Note this value is pulled from the nav_graph.xml
+private const val SERIES_TYPE = "seriesType"
+
+// TODO: Show a "loading" screen, which should be a list of shimmering items
```

Comment:
Still need to handle this before this can be merged

Reply:
I don't think the screen will be visible for enough time that this should be needed... maybe just show nothing while it loads?

Reply:
Will do this later on so this can get merged instead of sitting here